org.qtproject.qtcreator.yaml

```diff
@@ -0,0 +1,85 @@
+app-id: org.qtproject.qtcreator # id "org.qt-project.qtcreator" cannot be used
+runtime: org.kde.Sdk
+runtime-version: "6.2"
+sdk: org.kde.Sdk
+command: qtcreator
+# Qt Creator already provides desktop launchers and AppData metadata
+# but since we cannot use the original AppID, we have to rename the original files
+rename-desktop-file: org.qt-project.qtcreator.desktop
+rename-appdata-file: org.qt-project.qtcreator.appdata.xml
+rename-icon: QtProject-qtcreator
+finish-args:
+# Qt Creator crashes GNOME Shell 3.36 on Wayland
+# - --socket=wayland
+# - --socket=fallback-x11
+- --socket=x11
+- --share=ipc
+- --share=network
+- --device=dri
+- --filesystem=host
+cleanup:
+- /include
+- /lib/arch
+- /lib/cmake
+- /lib/libQt6Core5Compat.prl
+- /lib/metatypes
+- /lib/qtcreator/objects-RelWithDebInfo
+- /lib/x86_64-linux-gnu
+- /mkspecs
+- /modules
+- /qml
+- "*.a"
+cleanup-commands:
+- mv /app/lib/*-linux-gnu/* /app/lib/
+
+modules:
+# addition Qt modules that are not part of the KDE SDK
+- name: qt5compat
+  sources:
+  - type: git
+    url: git://code.qt.io/qt/qt5compat.git
+    tag: v6.2.1
+  buildsystem: cmake-ninja
+  builddir: true
+  post-install:
+  # move cmake files into a known path because we cannot determine the arch triplet
+  # (i.e. "x86_64-linux-gnu") using shell commands or environment variables
+  - mkdir /app/lib/arch/
+  - mv /app/lib/*-linux-gnu/cmake /app/lib/arch/
+
+- name: QtCreator
+  sources:
+  - type: git
+    url: git://code.qt.io/qt-creator/qt-creator.git
+    tag: v6.0.1
+  buildsystem: simple
+  build-commands:
+  # run cmake manually to install the "Devel" component for plugin development
+  - cmake -GNinja -B ${FLATPAK_BUILDER_BUILDDIR} -DCMAKE_BUILD_TYPE=RelWithDebInfo -DQt6Core5Compat_DIR=/app/lib/arch/cmake/Qt6Core5Compat/
+  - cmake --build ${FLATPAK_BUILDER_BUILDDIR}
+  - cmake --install ${FLATPAK_BUILDER_BUILDDIR} --prefix=/app
+  - cmake --install ${FLATPAK_BUILDER_BUILDDIR} --prefix=/app --component Devel
```

Comment:
what's wrong with using `buildsystem: cmake-ninja`

Reply:
I have to pass `--component Devel` to the last install step. I haven't found an alternative way to do this with flatpak-biulder.

Reply:
`make-install-args`

Reply:
> `make-install-args`

This would only work for a single install step, right? Passing `--component Devel` will only install the `Devel` component and not the actual program. This is why I have to call the install step twice with different arguments. How would I have the install step called twice with different components?

Also, the documentation says "An array of extra arguments that will be passed to make install". Does this still apply to `cmake --install` and/or `ninja install`?

Reply:
you can also run it in the `post-install` phase.

Reply:
If I use `cmake-ninja` with
```yaml
post-install:
  - cmake --install ${FLATPAK_BUILDER_BUILDDIR} --prefix=/app --component Devel
```
then the "normal installation" of the runtime and executables works, but the post-install step fails with only `CMake Error: Error processing file: /run/build/QtCreator/cmake_install.cmake`.

The verbose flatpak-builder output `FB: Running: flatpak build --die-with-parent --env=FLATPAK_BUILDER_BUILDDIR=/run/build/QtCreator [...] /bin/sh -c 'cmake --install ${FLATPAK_BUILDER_BUILDDIR} --prefix=/app --component Devel'` looks normal and verbose make output also does not provide additional information.

Is it possible to halt the build process at this stage, enter a shell and try to debug the command by changing the generated `cmake_install.cmake`?

Reply:
The same `post-install` step works if I am calling cmake manually in the `simple` build system like before, excluding what was split into the post-install step:
```yaml
buildsystem: simple
build-commands:
- cmake -GNinja -B ${FLATPAK_BUILDER_BUILDDIR} -DCMAKE_BUILD_TYPE=RelWithDebInfo -DQt6Core5Compat_DIR=/app/lib/arch/cmake/Qt6Core5Compat/ -DBUILD_PLUGINS_BY_DEFAULT:BOOL=OFF -DBUILD_EXECUTABLE_BUILDOUTPUTPARSER:BOOL=OFF -DBUILD_EXECUTABLE_CPASTER:BOOL=OFF -DBUILD_EXECUTABLE_CPLUSPLUS-KEYWORDGEN:BOOL=OFF -DBUILD_EXECUTABLE_QML2PUPPET:BOOL=OFF -DBUILD_LIBRARY_AGGREGATION=ON -DBUILD_LIBRARY_EXTENSIONSYSTEM=ON -DBUILD_LIBRARY_UTILS=ON
- cmake --build ${FLATPAK_BUILDER_BUILDDIR}
- cmake --install ${FLATPAK_BUILDER_BUILDDIR} --prefix=/app
post-install:
- cmake --install ${FLATPAK_BUILDER_BUILDDIR} --prefix=/app --component Devel
```
But using `cmake-ninja`:
```yaml
buildsystem: cmake-ninja
builddir: true
config-opts:
- -DCMAKE_BUILD_TYPE=RelWithDebInfo
- -DQt6Core5Compat_DIR=/app/lib/arch/cmake/Qt6Core5Compat/
[...]
post-install:
- cmake --install ${FLATPAK_BUILDER_BUILDDIR} --prefix=/app --component Devel
```
will fail.

So generally, this `post-install` step would work, but calling cmake manually behaves differently to the `cmake-ninja` procedure.